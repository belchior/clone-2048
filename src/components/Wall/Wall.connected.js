import { connect } from 'react-redux';
import { Wall } from './Wall';

export const mapStateToProps = (state) => {
  return { hardMode: state.hardMode };
};

export const mapDispatchToProps = () => {
  return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(Wall);
