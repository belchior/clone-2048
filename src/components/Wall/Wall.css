@import '../../static/vars.css';

@media all {
  .Wall {
    display: flex;
    flex-flow: row wrap;
    flex: 1 0 100%;
    border: 1px solid var(--gray1);
    width: 100%;
    max-width: 32rem;
    max-height: 32rem;
  }
  .Wall.hardMode,
  .Wall.hardMode .Block {
    border: 0;
    color: transparent;
  }
}
