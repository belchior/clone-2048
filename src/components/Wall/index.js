export * from './Wall';
export * from './Wall.connected';
export { default } from './Wall.connected';
