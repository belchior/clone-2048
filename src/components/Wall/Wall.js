import React from 'react';
import PropTypes from 'prop-types';

import './Wall.css';
import { Block } from '../Block';

export const Wall = ({ hardMode }) => (
  <div className={'Wall ' + (hardMode ? 'hard-mode' : '')}>
    <Block value={2}></Block>
    <Block value={4}></Block>
    <Block value={8}></Block>
    <Block value={16}></Block>
    <Block value={4096}></Block>
    <Block value={0}></Block>
    <Block value={0}></Block>
    <Block value={32}></Block>
    <Block value={2048}></Block>
    <Block value={0}></Block>
    <Block value={0}></Block>
    <Block value={64}></Block>
    <Block value={1024}></Block>
    <Block value={512}></Block>
    <Block value={256}></Block>
    <Block value={128}></Block>
  </div>
);

Wall.propTypes = {
  hardMode: PropTypes.bool,
};
