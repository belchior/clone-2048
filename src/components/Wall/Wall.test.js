import React from 'react';
import { Wall } from './Wall';
import renderer from 'react-test-renderer';

it('should render Wall without crashing', () => {
  let tree = renderer.create(<Wall />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render Wall without crashing with parameters', () => {
  let tree = renderer.create(<Wall hardMode={true}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
