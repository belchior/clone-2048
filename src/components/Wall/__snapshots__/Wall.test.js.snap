// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Wall without crashing 1`] = `
<div
  className="Wall "
>
  <div
    className="Block block-2"
  >
    2
  </div>
  <div
    className="Block block-4"
  >
    4
  </div>
  <div
    className="Block block-8"
  >
    8
  </div>
  <div
    className="Block block-16"
  >
    16
  </div>
  <div
    className="Block block-4096"
  >
    4096
  </div>
  <div
    className="Block block-0"
  >
    0
  </div>
  <div
    className="Block block-0"
  >
    0
  </div>
  <div
    className="Block block-32"
  >
    32
  </div>
  <div
    className="Block block-2048"
  >
    2048
  </div>
  <div
    className="Block block-0"
  >
    0
  </div>
  <div
    className="Block block-0"
  >
    0
  </div>
  <div
    className="Block block-64"
  >
    64
  </div>
  <div
    className="Block block-1024"
  >
    1024
  </div>
  <div
    className="Block block-512"
  >
    512
  </div>
  <div
    className="Block block-256"
  >
    256
  </div>
  <div
    className="Block block-128"
  >
    128
  </div>
</div>
`;

exports[`should render Wall without crashing with parameters 1`] = `
<div
  className="Wall hard-mode"
>
  <div
    className="Block block-2"
  >
    2
  </div>
  <div
    className="Block block-4"
  >
    4
  </div>
  <div
    className="Block block-8"
  >
    8
  </div>
  <div
    className="Block block-16"
  >
    16
  </div>
  <div
    className="Block block-4096"
  >
    4096
  </div>
  <div
    className="Block block-0"
  >
    0
  </div>
  <div
    className="Block block-0"
  >
    0
  </div>
  <div
    className="Block block-32"
  >
    32
  </div>
  <div
    className="Block block-2048"
  >
    2048
  </div>
  <div
    className="Block block-0"
  >
    0
  </div>
  <div
    className="Block block-0"
  >
    0
  </div>
  <div
    className="Block block-64"
  >
    64
  </div>
  <div
    className="Block block-1024"
  >
    1024
  </div>
  <div
    className="Block block-512"
  >
    512
  </div>
  <div
    className="Block block-256"
  >
    256
  </div>
  <div
    className="Block block-128"
  >
    128
  </div>
</div>
`;
